zorch/lnp/exact.py

```
@@ -457,39 +544,40 @@ def require_no_wraparound(
+        a statement the caller wrote down and this object never sees."""
         if binary_cols < 0:
             raise ValueError(
                 f"exact: the binary-column width cannot be negative, got "
                 f"{binary_cols!r}"
             )
+        if leg.scheme.ring is not self.ring:
+            raise ValueError(
+                "exact: the leg and this statement must hold one ring — a "
+                "bound proved over a different modulus prices nothing here"
+            )
+        # Raises for Fig. 10's ℓ∞ leg, which proves no ℓ2 bound to price.
```

Ruff (0.16.2)

[warning] Comment contains ambiguous `ℓ` (SCRIPT SMALL L). Did you mean `l` (LATIN SMALL LETTER L)?

(RUF003)